package.json

```diff
@@ -11,20 +11,25 @@
         "url": "git://github.com/lokalise/node-core.git"
     },
     "license": "Apache-2.0",
-    "files": ["dist/**", "LICENSE", "README.md"],
-    "main": "dist/index.js",
-    "types": "dist/index.d.ts",
-    "type": "commonjs",
+    "files": ["dist"],
+    "type": "module",
+    "exports": {
+        "./package.json": "./package.json",
+        ".": {
+            "import": "./dist/index.js",
+            "default": "./dist/index.cjs"
+        }
+    },
     "publishConfig": {
         "access": "public"
     },
     "scripts": {
-        "build": "tsc",
+        "build": "tsup",
```

Comment:
Just out of curiosity, and most likely I lack some knowledge 😓 so please bear with me.
AFAIK the current build system works with `cjs` and `esm` services and we are migrating our services to ESM, could make sense to wait until all services are migrated and then just go with ESM?

Sorry if the question is very stupid 🙏

Reply:
Previously, we were building to `cjs` only, with tsup we are doing dual-build. In the dist there are .js (ESM) and .cjs (CJS) now. Besides that, the current approach prevents from accessing subpaths in imports.
Once all services are rewritten to ESM we can get rid of CJS entirely.

Reply:
Considering that a lot of stuff depend on node-core, including some external libraries like message-queue-toolkit, which are used outside of just Lokalise, and forcing all of them to go ESM-only wouldn't be particularly nice. I'd say that dual publish makes sense until at least Node 24, which will finally support ESM imports in CJS world.

Reply:
ESM doesn't work without dual-publish. Based on local test
```
// tmp.mjs
import { removeDuplicates } from '@lokalise/node-core'

console.log(removeDuplicates([1, 1]))
```
Error [ERR_PACKAGE_PATH_NOT_EXPORTED]: No "exports" main defined in node_modules/@lokalise/node-core/package.json imported from polyglot-service/tmp.mjs

Reply:
@kibertoad yeah, we shouldn't go ESM only before Node 24 LTS

Reply:
I see! thanks for explanation it makes a lot of sense 🙏